### PR TITLE
Collect arguments for application session stickiness and update docs

### PR DESCRIPTION
**Pull request by heytrav**
##### SUMMARY

Additional attributes for elb target group to support application-based session cookies

##### ISSUE TYPE

- Feature Pull Request

##### COMPONENT NAME

elb_target_group
##### ADDITIONAL INFORMATION

See https://github.com/ansible-collections/community.aws/issues/513

### CODE REVIEW

File: plugins/modules/elb_target_group.py [hunk 1]
```diff
@@ -102,9 +102,21 @@
       - The time period, in seconds, during which requests from a client should be routed to the same target. After this time period expires, the load
         balancer-generated cookie is considered stale. The range is 1 second to 1 week (604800 seconds).
     type: int
+  stickiness_app_cookie_duration:
+    description:
+      - The time period, in seconds, during which requests from a client
+        should be routed to the same target. After this time period expires,
+        the application-generated cookie is considered stale. The range is 1 second to 1 week (604800 seconds).
+    type: int
```
Comment (tremble) [inline plugins/modules/elb_target_group.py:110]:
```suggestion
    type: int
    version_added: 1.5.0
```

File: plugins/modules/elb_target_group.py [hunk 2]
```diff
@@ -102,9 +102,21 @@
...
+  stickiness_app_cookie_name:
+    description:
+      - The name of the application cookie. Required if C(stickiness_type) is
+        C(app_cookie).
+    type: str
```
Comment (tremble) [inline plugins/modules/elb_target_group.py:115]:
```suggestion
    type: str
    version_added: 1.5.0
```

**Review by tremble — CHANGES_REQUESTED** [review]
@heytrav thanks for taking the time to open this PR.

The change looks sensible and matches with the upstream docs.  Please would you add a changelog entry and the "version_added" changes.

https://docs.ansible.com/ansible/latest/community/development_process.html#changelogs-how-to

File: plugins/modules/elb_target_group.py (hunk 2 above)
Comment (heytrav) [reply plugins/modules/elb_target_group.py:116]: Ok, added

**Review by tremble — APPROVED** [review]

File: plugins/modules/elb_target_group.py [hunk 3]
```diff
@@ -102,9 +102,23 @@
...
+    description:
+      - The time period, in seconds, during which requests from a client
+        should be routed to the same target. After this time period expires,
+        the application-generated cookie is considered stale. The range is 1 second to 1 week (604800 seconds).
+    type: int
+    version_added: 1.5.0
+  stickiness_app_cookie_name:
+    description:
+      - The name of the application cookie. Required if C(stickiness_type) is
+        C(app_cookie).
```
Comment (tremble) [inline plugins/modules/elb_target_group.py:115]:
```suggestion
      - The name of the application cookie. Required if I(stickiness_type=app_cookie).
```

Just a niggle I'll merge, but we use `I(...)` for parameters and `C(...)` for values.

File: plugins/modules/elb_target_group.py [hunk 4]
```diff
@@ -102,9 +102,23 @@
...
+    type: str
+    version_added: 1.5.0
   stickiness_type:
     description:
       - The type of sticky sessions.
+      - C(lb_cookie), C(app_cookie) or C(source_ip)
```
Comment (tremble) [inline plugins/modules/elb_target_group.py:121]:
```suggestion
      - Valid values are C(lb_cookie), C(app_cookie) or C(source_ip)
```

**Pull request merged**